backends/bmv2/psa_switch/psaSwitch.cpp

```diff
@@ -702,6 +737,29 @@ void ExternConverter_DirectCounter::convertExternInstance(
         jctr->emplace("is_direct", true);
         jctr->emplace("binding", it->second->controlPlaneName());
         ctxt->json->counters->append(jctr);
+
+        // Adding direct counter to EXTERN_INSTANCES
+
+        auto extern_obj = new Util::JsonObject();
+        extern_obj->emplace("name", name);
+        extern_obj->emplace("id", nextId("extern_instances"));
+        extern_obj->emplace("source_info", inst->sourceInfoJsonObj());
+        ctxt->json->externs->append(extern_obj);
+        Util::JsonArray *arr = ctxt->json->insert_array_field(extern_obj, "attributes");
+
+        // Direct Counter only has a single argument, which is psa metadata
+        // converter in conversion context will handle this for us
+        auto tp = eb->findParameterValue("type");
+        CHECK_NULL(tp);
+        if (!tp->is<IR::Declaration_ID>()) {
```

Comment:
Perhaps you can make this if !tp || !tp->is...

Reply:
Changed in both places.